robot/server/server.go

```diff
@@ -417,25 +418,38 @@ func (s *Server) Log(ctx context.Context, req *pb.LogRequest) (*pb.LogResponse,
 	}
 	log := req.Logs[0]
 
+	var fields []zap.Field
+	for _, fieldP := range log.Fields {
+		fieldPJSON, err := fieldP.MarshalJSON()
```

Comment:
I don't _love_ going from `zap.Field` -> proto -> JSON -> `zap.Field` 🤷🏻 .

Reply:
I was expecting we'd be building up a slice of anys, `fields []any` that would go into a `w` log call, e.g: `Debugw(msg, fields...)`.

Where the elements of `fields` would alternate between keys and values:
```
"moduleLogTS", "2024-02-26T14:27:09",
"voltage", 0.1000000037,
"total_rotations", 365,
```

I'm sure there's lots of challenges getting this all plumbed through. I don't think I've appreciated the challenges though. We should maybe look over some of this together so I can give some better feedback.

Reply:
Yeah I wish we could build up a slice of `[]any`; I don't quite know how to do that without reimplementing the encoding parts of zap's library. Basically, a [`zap.Field`](https://pkg.go.dev/go.uber.org/zap#Field) is a type alias for a `zapcore.Field` which looks like:
```
type Field struct {
	Key       string
	Type      FieldType
	Integer   int64
	String    string
	Interface interface{}
}
```
I thought it was easier to `Desugar` the logger and pass `Field`s directly to the logger rather than try to convert `Field`s to a native golang type to log them with a sugared logger and something like `Debugw`. I'm happy to meet/look things over if desired! But, I suspect this code will inevitably look a bit weird if we're trying to send `zap.Field`s over the wire.

Reply:
Built up a slice of `[]any`; see new `FieldKeyAndValueFromProto` function.